0_From_Code_First_To_First_Principles/1_code_first/assignment/src/controllers/users.ts

```diff
@@ -0,0 +1,80 @@
+import type { RequestHandler } from 'express';
+
+import { Context } from '../interfaces';
+import { UserDTO, UserDb, createUser, getUserByEmail, updateUser } from '../models';
+
+const userDbToUser = (u: UserDb): UserDTO => ({ id: u.id, email: u.email, username: u.username, firstName: u.firstname, lastName: u.lastname });
+
+function generateRandomPassword(length = 10) {
+  const charset = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789';
+  let password = '';
+  for (let i = 0; i < length; i++) {
+    const randomIndex = Math.floor(Math.random() * charset.length);
+    password += charset.charAt(randomIndex);
+  }
+  return password;
+}
+
+export const userCreate: (context: Context) => RequestHandler = (context) => async (req, res) => {
```

Comment:
This signature was quite hard to read. Maybe it could be simplified by creating a distinct type, something like this:

```
type RequestHandlerFactoryFn = (context: Context) => RequestHandler
```
... and then ...
```
export const userCreate: RequestHandlerFactoryFn = ...
export const userEdit: RequestHandlerFactoryFn = ...
// and so forth ...
```